testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java

```diff
@@ -42,13 +45,74 @@
 public abstract class BaseConnectorSmokeTest
         extends AbstractTestQueryFramework
 {
+    private static final Logger LOG = Logger.get(BaseConnectorSmokeTest.class);
+
     protected static final List<TpchTable<?>> REQUIRED_TPCH_TABLES = ImmutableList.of(NATION, REGION);
 
     protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
     {
         return connectorBehavior.hasBehaviorByDefault(this::hasBehavior);
     }
 
+    /**
+     * Get a {@code SqlExecutor} that allows table and schema creation.
+     *
+     * <p>If an empty optional is returned, tests will attempt to use {@link
+     * #getQueryRunner()} instead, if the connector allows table/schema
+     * creation.
+     *
+     * <p>To override behavior for individual types of statements, see
+     * {@link #createTableAs(String, String)},
+     * {@link #createTableAs(String, String)}, and {@link #dropTable(String)}.
+     */
+    protected Optional<SqlExecutor> getSqlExecutor()
+    {
+        return Optional.empty();
```

Comment:
I guess the `SqlExecutor` is expected to accept Trino syntax (otherwise it's not reusable), so i cannot think of a situation where this could be anything else than `getQueryRunner()::execute`

(in any case, it should not be called "onRemoteDatabase", because the remote database cannot be expected to accept Trino syntax)